code/iomethods.py

```diff
@@ -518,5 +532,340 @@ def decode_time_from_string(timeString):
+        Assumptions:: fileName is in the format: merg_YYYYMMDDHH_4km-pixel
+
+    '''
+
+    newFilePath = os.path.join(dirName, fileName + '.nc')
+    ncdf = netCDF4.Dataset(newFilePath, "w", format="NETCDF4")
+
+    # Set global attributes, dimensions, and lastly, each variable
+    ncdf.setncatts(globalAttrDict)
+
+    for nameOfDimension, size in dimensionsDict.iteritems():
+        ncdf.createDimension(nameOfDimension, size)
+
+    ncdf.createVariable(lonDict['name'], lonDict['dataType'], lonDict['dimensions'])
+    ncdf.variables[lonDict['name']].setncattr('units', lonDict['units'])
+    ncdf.variables[lonDict['name']].setncattr('long_name', lonDict['long_name'])
+
+    ncdf.createVariable(latDict['name'], latDict['dataType'], latDict['dimensions'])
+    ncdf.variables[latDict['name']].setncattr('units', latDict['units'])
+    ncdf.variables[latDict['name']].setncattr('long_name', latDict['long_name'])
+
+    ncdf.createVariable(timeDict['name'], timeDict['dataType'], timeDict['dimensions'])
+
+    # Parse time from filename. Assumes the filename is in the format 'merg_xxxxxxxxxx_4km-pixel'
+    dateFromFileName = fileName[5:15]
+    dateAsDateTime = datetime.strptime(dateFromFileName, '%Y%m%d%H')
+
+    ncdf.variables['time'].setncattr('units', 'hours since ' + str(dateAsDateTime.year) + '-' + str(dateAsDateTime.month) +
+                                     '-' + str(dateAsDateTime.day) + ' ' + str(dateAsDateTime.hour))
+
+    ncdf.createVariable(ch4Dict['name'], ch4Dict['dataType'], ch4Dict['dimensions'])
+    ncdf.variables[ch4Dict['name']].setncattr('long_name', ch4Dict['long_name'])
+    ncdf.variables[ch4Dict['name']].setncattr('time_statistic', ch4Dict['time_statistic'])
+    ncdf.variables[ch4Dict['name']].setncattr('missing_value', ch4Dict['missing_value'])
+
+    ncdf.variables['longitude'][:] = lonDict['values']
+    ncdf.variables['latitude'][:] = latDict['values']
+    ncdf.variables['ch4'][:,:,:] = ch4Dict['values']
+
+    ncdf.close()
+
+    return newFilePath
+
+
+def _read_merg_file(filepath, shape, offset):
+    '''
+        Purpose:: Read merg data from a binary file and put it into a numPy array
+
+        Input:: filepath - The path to the merg binary file
+                Shape - The shape we want the data to be in
+                Offset - Data is sometimes scaled to fit by reducing the value, so add it back if necessary
+
+        Output:: data - A numPy array with datatype np.float with the shape that was passed in
+
+    '''
+
+    mergFile = open(filepath, 'rb')
+    mergArray = np.fromfile(mergFile, dtype=np.uint8, count=-1)  # count=-1 means read entire file
+    data = mergArray.astype(np.float).reshape(shape)
+    data += offset
+
+    mergFile.close()
+
+    return data
+
+def _get_valid_coordinate(coordinateName, expectedLow, expectedHigh):
+    try:
+        input = float(raw_input(coordinateName + " passed to read_netCDF_to_array "
```

Comment:
input should be avoided as variable name. Although it isn't a reserved word, it conflicts with the built-in Python function